src/sentry/models/repository.py

```diff
@@ -97,3 +101,4 @@ def handle_exception(exc):
 
 
 pending_delete.connect(on_delete, sender=Repository, weak=False)
+pre_delete.connect(delete_pending_deletion_option, sender=Repository, weak=False)
```

Comment:
I hesitated on using a signal as they do not always fire the way we do our deletions. Would an asynchronous task to purge the options table work better instead?

Reply:
I think this is fine in this context.